### ISSUE TITLE
NSIS - Poor differential updater performance

### ISSUE THREAD

**Issue by burnhamup** [electron-userland/electron-builder#6265]
* **Electron-Builder Version**: 22.14.2

* **Node Version**: 14.17.5

* **Electron Version**: 13.3.0
* **Electron Type (current, beta, nightly)**: current

* **Electron-Updater Version**: 4.6.0

* **Target**: Nsis

I finally got the differential downloader working with my update system. But I'm realizing that the installers that NSIS produces don't perform well with the differential downloader. After comparing the blockmaps from two versions of my app, 97% of the data is different and needs to be downloaded. From my logs: 

```[2021-09-17 15:59:52.654] [info]  Full: 65,349.73 KB, To download: 63,610.3 KB (97%)```

 I  compared the actual install folders of these two versions and ended up with a much closer comparison
```
∙ Hashing build-205
√ 205.32 MiB (117 files, 19 dirs, 0 symlinks) @ 79.90 MiB/s

∙ Diffing build-206
√ 205.35 MiB (117 files, 18 dirs, 0 symlinks) @ 75.21 MiB/s

√ Re-used 99.21% of old, added 1.62 MiB fresh data
√ 626.61 KiB patch (0.30% of the full size) in 2.7305017s
```

My theory is that the compression done by the installer destroys the benefits of doing a differential download. Comparing the inner app-64.7z of the two installers shows no blocks that can be reused. 

## Ideas
One solution could be to remove the differential download feature. If this is happening across the board, then it's not really saving anyone bandwidth. It's been tricky to debug issues with it, and may be more trouble than its worth. 

Another solution could be to investigate alternate settings for NSIS. Maybe a different compression option would yield much better patches. I imagine there would be some tradeoff here - an uncompressed installer for my application would weigh closer to 200MB than 60MB, but I would potentially have 1MB patches. 

I wanted to raise the issue to make the maintainers aware of the current limitations of the differential downloader. I'll look into alternate compression settings and report back with what I learn.

**Comment by burnhamup** [comment]
Found some relevant information while poking around:

https://github.com/electron-userland/electron-builder/issues/1523 
and https://github.com/electron-userland/electron-builder/commit/fcae4c2d035f8329e51f22435d5dd79fe0f4e5de 
reference a big addition of differential downloads for the nsis-web target

Some of the comments imply that this only works for nsis-web and not nsis

https://github.com/electron-userland/electron-builder/issues/2217 and https://github.com/electron-userland/electron-builder/commit/acf783d2d001ffd0eb5a25ced402818dcb8ed33e 
added support for regular nsis. 

I'm looking at some of the config options here, like compression: 'store' to see what effects they have. 

Given that this might be working for nsis-web and other targets, I no longer want to remove the feature.

**Comment by burnhamup** [comment]
I dug deeper into the NsisTarget.js and ran more tests on my machine. 

Everything is working as expected. The compression options from configureDifferentialAwareArchiveOptions() are taking effect.
 Producing builds from my dev machine results in installers with very small patches between them. I also ruled out any issues with code signing. 

I think I've identified my issue. Our unsigned builds are prepared with a Linux server, and our signed builds are prepared with a Windows server. When I update from a signed build to another signed build, I get the patch performance I expect. 

My hunch is that the 7zip compression probably differs between Win/Linux and I certainly can't expect this electron-builder to do anything about that.
